src/react/components/searchField.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState, createRef, useEffect } from "react";
+import { X, Search } from "react-hero-icon/solid";
+import SearchAutocomplete from "./searchAutocomplete";
+import UseEscape from "../hooks/useEscape";
+import UseOnClickOutside from "../hooks/useOnClickOutside";
+
+const SearchField = React.forwardRef(
+  (
+    {
+      onSubmit,
+      onReset,
+      onInput,
+      autocompleteSuggestions,
+      defaultValue,
+      inputClasses,
+      wrapperClasses,
+      submitBtnClasses,
+      clearBtnClasses,
+      autocompleteLinkClasses,
+      autocompleteLinkFocusClasses,
+      autocompleteContainerClasses,
+      placeholder,
+      clearOnEscape,
+    },
+    ref
+  ) => {
+    const [query, setQuery] = useState(defaultValue || "");
+    const [showAutocomplete, setShowAutocomplete] = useState(false);
+    const [selectedSuggestion, setSelectedSuggestion] = useState(null);
+    const inputWrapper = createRef();
+    const inputRef = ref || createRef();
+
+    const submitHandler = (e) => {
+      e.preventDefault();
+      setShowAutocomplete(false);
+      onSubmit(query);
+    };
+
+    const inputHandler = (e) => {
+      setQuery(e.target.value);
+      onInput(e.target.value);
+      setShowAutocomplete(true);
+    };
+
+    const clearHandler = (e) => {
+      if (e) {
+        e.preventDefault();
+      }
+      setQuery("");
+      setShowAutocomplete(false);
+      setSelectedSuggestion(null);
+      onReset();
+    };
+
+    const selectSuggestion = (e, suggestion) => {
+      e.preventDefault();
+      setQuery(suggestion);
+      setShowAutocomplete(false);
+      setSelectedSuggestion(null);
+      onSubmit(suggestion);
+    };
+
+    useEffect(() => {
+      setQuery(defaultValue);
+    }, [defaultValue]);
+
+    UseOnClickOutside(inputWrapper, () => {
+      setShowAutocomplete(false);
+    });
+
+    const handleArrowKeys = (e) => {
+      if (e.key === "ArrowDown") {
+        e.preventDefault();
+        setSelectedSuggestion(selectedSuggestion + 1);
+      } else if (e.key === "ArrowUp") {
+        e.preventDefault();
+        setSelectedSuggestion(selectedSuggestion - 1);
+      } else if (
+        e.key === "Enter" &&
+        autocompleteSuggestions[selectedSuggestion]
+      ) {
+        selectSuggestion(e, autocompleteSuggestions[selectedSuggestion].query);
+      }
+    };
+
+    UseEscape(() => {
+      if (clearOnEscape && document.activeElement === inputRef.current) {
+        clearHandler();
+      }
+    });
+
+    return (
+      <div>
+        <form onSubmit={submitHandler}>
+          <div className="flex items-center">
+            <span className="" />
```

Comment:
What is this empty span for?

Reply:
I'll confirm if this is needed and remove if not.